### PR TITLE
[PodTargetInstaller] Set deployment target to maximum spec deployment…

### LINKED ISSUE #3454: respect the deployment_target of a dependency when building
XCode 6.3 has started to use the deployment target of a library when performing static analysis.

Currently CocoaPods overrides the deployment target of the podspec with the platform directive in each target. For example:

```
target 'MyApp' do
    platform :ios, '6.1'
    pod 'CorePlot' # <-- has a deployment_target of 5.0
    pod 'AFNetworking'
end
```

The library `libPods-MyApp-CorePlot.a` will be built with a deployment target of 6.1 instead of 5.0. This causes a lot of warnings in static analysis because CorePlot has shims in the code that make the library work on ios 5.0. However, XCode realizes those shims are no longer necessary if deploying to 6.1+ and emits warnings.

A possible work around for this was to create multiple targets with their own deployment targets:

```
target 'MyApp-50' do
    platform :ios, '5.0'
    link_with 'MyApp'
    pod 'CorePlot'
end

target 'MyApp-61' do
    platform :ios, '6.1'
    link_with 'MyApp'
    pod 'AFNetworking'
end
```

Unfortunately each of these targets has its own xcconfig and CocoaPods doesn't merge these xcconfigs into one, causing issues. So CocoaPods doesn't seem to support two pods linking against the same entry point (`MyApp`).

The correct solution (it seems to me) is that the deployment target should be used from the podspec when building the library. For example, CorePlot has `spec.ios.deployment_target '5.0'`, so when building `libPods-MyApp-CorePlot.a` the deployment target should be set to 5.0. This seems reasonable to me since we're talking about configuring the dependencies, not `MyApp` (which has its own xcodeproj that I configure). Everything works fine if I manually click to set the deployment target on there in the generated xcodeproj, but obviously this gets wiped out when I run `pod update` again.

### PR DESCRIPTION
**Pull request by segiddins**
… target

Closes #3454.

- [x] CHANGELOG
- [x] Specs

### CODE REVIEW

File: lib/cocoapods/installer/target_installer/pod_target_installer.rb [hunk 1]
```diff
@@ -285,6 +285,15 @@ def custom_module_map
         @custom_module_map ||= target.file_accessors.first.module_map
       end
 
+      # The deployment target for the pod target, which is the maximum of all
+      # the deployment targets for the current platform of the target.
+      #
+      # @return [String] The deployment target.
+      #
+      def deployment_target
+        target.specs.map { |spec| spec.deployment_target(target.platform) }.max
```
Comment (segiddins) [inline lib/cocoapods/installer/target_installer/pod_target_installer.rb]: this needs to `||` with `PLATFORM_DEFAULTS[target.platform]`

File: lib/cocoapods/installer/target_installer/pod_target_installer.rb [hunk 2]
```diff
@@ -285,6 +285,16 @@ def custom_module_map
...
       end
 
+      # The deployment target for the pod target, which is the maximum of all
+      # the deployment targets for the current platform of the target.
+      #
+      # @return [String] The deployment target.
+      #
+      def deployment_target
+        default = Podfile::TargetDefinition::PLATFORM_DEFAULTS[target.platform.name]
+        target.specs.map { |spec| spec.deployment_target(target.platform) || default }.max
```
Comment (floere) [inline lib/cocoapods/installer/target_installer/pod_target_installer.rb]: Won't this fail in the standard problem case `ruby -e 'p ["10.10", "10.9"].max'`?

Comment (segiddins) [inline lib/cocoapods/installer/target_installer/pod_target_installer.rb]: Ah you're right. Need to wrap into a Pod::Version

**Comment by segiddins** [comment]
Will merge once CI runs

**Pull request merged**